drivers/include/periph/flashpage.h

```diff
@@ -27,6 +23,20 @@
  *              around 10K times), so using this interface in some kind of loops
  *              can damage you MCU!
  *
+ * @section     sec_pm (Low-) power implications
+ *
+ * The flashpage driver implementation **should** make sure, that the CPU uses
+ * no additional energy while the flashpage driver is active. This means, that
+ * any particular CPU peripherals used for reading and writing flash pages
+ * **should** be disabled before the read and write functions return.
+ *
```

Comment:
there is a typo here, it should of course be made sure that no power is used when the driver is **inactive**...

Not sure I understand your undesired code block, though...